### PR TITLE
TextCursor

### PR DESCRIPTION
**Pull request by stephengillie**
- [ ] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-dsc).
- [ ] This pull request is related to an issue.

-----

### CODE REVIEW

**Comment by stephengillie** [comment]
Still getting an error, but this is as far as I'm getting tonight. 

```PowerShell
 [-] TextCursor.TextCursorIndicatorStatus. 65ms (64ms|1ms)
 RuntimeException: The variable '$global:NTAccessibilityRegistryPath' cannot be retrieved because it has not been set.
 at GetTextCursorIndicatorStatus, C:\Users\Stephen.Gillie\Documents\PowerShell\Modules\Microsoft.Windows.Setting.Accessibility\Microsoft.Windows.Setting.Accessibility.psm1:452
 at Get, C:\Users\Stephen.Gillie\Documents\PowerShell\Modules\Microsoft.Windows.Setting.Accessibility\Microsoft.Windows.Setting.Accessibility.psm1:509
[
```

File: resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1 [hunk 1]
```diff
@@ -419,6 +421,179 @@ class Audio
...
+#>
+[DSCResource()]
+class TextCursor
+{
+    # Key required. Do not set.
+    [DscProperty(Key)] [string] $SID
+    [DscProperty()] [nullable[bool]] $TextCursorIndicatorStatus
+    [DscProperty()] [int] $TextCursorIndicatorSize
+    [DscProperty()] [int] $TextCursorIndicatorColor
+    [DscProperty()] [int] $TextCursorThickness
```
Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:441]: It's already assumed that these properties relate to `TextCursor` so these can be shortened. 
```suggestion
    [DscProperty()] [nullable[bool]] $Status
    [DscProperty()] [int] $Size
    [DscProperty()] [int] $Color
    [DscProperty()] [int] $Thickness

```

File: resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1 [hunk 2]
```diff
@@ -35,12 +35,14 @@ if ([string]::IsNullOrEmpty($env:TestRegistryPath)) {
     $global:ControlPanelAccessibilityRegistryPath= 'HKCU:\Control Panel\Accessibility\'
     $global:AudioRegistryPath = 'HKCU:\Software\Microsoft\Multimedia\Audio\'
     $global:PersonalizationRegistryPath = 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Themes\Personalize\'
+    $global:NTAccessibilityRegistryPath = 'HKCU:\Software\Microsoft\Windows NT\CurrentVersion\Accessibility\'
+    $global:CursorIndicatorAccessibilityRegistryPath = $global:AccessibilityRegistryPath + '\CursorIndicator'
```
Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:39]: I don't see much benefit to reusing an existing variable. I would just write it out explicitly. That way you don't run into issues with your tests having dependencies on a global variable that has not yet been defined.

File: resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1 [hunk 3]
```diff
@@ -419,6 +421,179 @@ class Audio
...
+    [DscProperty()] [nullable[bool]] $TextCursorIndicatorStatus
+    [DscProperty()] [int] $TextCursorIndicatorSize
+    [DscProperty()] [int] $TextCursorIndicatorColor
+    [DscProperty()] [int] $TextCursorThickness
+
+    static hidden [string] $TextCursorIndicatorStatusProperty = 'Configuration'
+    static hidden [string] $TextCursorIndicatorStatusValue = 'cursorindicator'
+    static hidden [string] $TextCursorIndicatorSizeProperty = 'IndicatorType'
+    static hidden [string] $TextCursorIndicatorColorProperty = 'IndicatorColor'
+    static hidden [string] $TextCursorThicknessProperty = 'CaretWidth'
```
Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:447]: Same as above, these can be shortened.
```suggestion
    static hidden [string] $StatusPropertyName = 'Configuration'
    static hidden [string] $StatusValue = 'cursorindicator'
    static hidden [string] $SizePropertyName = 'IndicatorType'
    static hidden [string] $ColorPropertyName = 'IndicatorColor'
    static hidden [string] $ThicknessPropertyName = 'CaretWidth'

```

File: resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1 [hunk 4]
```diff
@@ -419,6 +421,179 @@ class Audio
...
+
+
+    static [bool] GetTextCursorIndicatorStatus()
+    {
+		$TextCursorIndicatorStatusArguments = @{  Path = $global:NTAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorStatusProperty)}
```
Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:452]:
```suggestion
	$statusArgs = @{  Path = $global:NTAccessibilityRegistryPath; Name = ([TextCursor]::IndicatorStatusProperty)}

```

Fix alignment as well as update the arg names for here and below..

File: resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1 [hunk 5]
```diff
@@ -419,6 +421,179 @@ class Audio
...
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+			$TextCursorIndicatorStatusArguments = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorStatusProperty); }
+			$TextCursorThicknessArguments = @{ Path = $global:ControlPanelDesktopRegistryPath; Name = ([TextCursor]::TextCursorThicknessProperty); }
+			$TextCursorIndicatorSizeArguments = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorSizeProperty)}
+			$TextCursorIndicatorColorArguments = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::TextCursorIndicatorColorProperty)}
```
Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:547]: Move these inside their respective if statements. That way if the condition is not satisfied, these arguments won't be created.

Comment (ryfu-msft) [inline resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1:547]:
```suggestion
			$statusArgs = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([TextCursor]::StatusProperty); }
			$thicknessArgs = @{ Path = $global:ControlPanelDesktopRegistryPath; Name = ([TextCursor]::ThicknessProperty); }
			$sizeArgs = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::SizeProperty)}
			$colorArgs = @{  Path = $global:CursorIndicatorAccessibilityRegistryPath; Name = ([TextCursor]::ColorProperty)}

```

**Review by ryfu-msft — CHANGES_REQUESTED** [review]

**Review by ryfu-msft — APPROVED** [review]

**Pull request merged**